test/acceptance/features/steps/steps.py

```diff
@@ -510,3 +510,14 @@ def create_deployment(context, app_name, image_ref):
     app = App(app_name, context.namespace.name, image_ref)
     if not app.is_running():
         assert app.install() is True, "Failed to create deployment."
+
+
+@then(u'Secret "{secret_name}" does not have key "{secret_key}"')
+def check_key_in_secret(context, secret_name, secret_key):
+    time.sleep(10)
+    openshift = Openshift()
+    json_path = f'{{.data.{secret_key}}}'
+    if not openshift.get_resource_info_by_jsonpath("secrets", secret_name, context.namespace.name, json_path):
+        assert True
```

Comment:
You don't need to `assert True` that is like a `nop` or effectively empty line... The assertion here should be only for the `False` case, also the failure message should explain more precisely, what went wrong - something like:
```python
assert openshift.get_resource_info_by_jsonpath("secrets", secret_name, context.namespace.name, json_path) is None, f"Secret '{secret_name}' must not have key '{secret_key}'"
```

Reply:
Updated, PTAL.

Reply:
Thx, but actually there is already a step for secret not having a secret... `@given(u'Secret "{secret_name}" does not contain "{key}"')` - it is only for `@given` but it can be changed to `@step` to cover the `Then` as well...